presto-delta/src/main/java/com/facebook/presto/delta/DeltaSplitManager.java

```diff
@@ -0,0 +1,138 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.delta;
+
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.ConnectorSplitSource;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.connector.ConnectorPartitionHandle;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import io.delta.standalone.actions.AddFile;
+import io.delta.standalone.data.CloseableIterator;
+import org.apache.hadoop.fs.Path;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toMap;
+
+public class DeltaSplitManager
+        implements ConnectorSplitManager
+{
+    private final String connectorId;
+    private final DeltaConfig deltaConfig;
+    private final DeltaClient deltaClient;
+    private final TypeManager typeManager;
+
+    @Inject
+    public DeltaSplitManager(DeltaConnectorId connectorId, DeltaConfig deltaConfig, DeltaClient deltaClient, TypeManager typeManager)
+    {
+        this.connectorId = requireNonNull(connectorId, "connectorId is null").toString();
+        this.deltaConfig = requireNonNull(deltaConfig, "deltaConfig is null");
+        this.deltaClient = requireNonNull(deltaClient, "deltaClient is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle handle,
+            ConnectorSession session,
+            ConnectorTableLayoutHandle layout,
+            SplitSchedulingContext splitSchedulingContext)
+    {
+        DeltaTableLayoutHandle layoutHandle = (DeltaTableLayoutHandle) layout;
+        return new DeltaSplitSource(session, layoutHandle.getTable());
+    }
+
+    private class DeltaSplitSource
+            implements ConnectorSplitSource
+    {
+        private final DeltaTable deltaTable;
+        private final CloseableIterator<AddFile> fileIterator;
+        private final int maxBatchSize;
+
+        DeltaSplitSource(ConnectorSession session, DeltaTableHandle deltaTableHandle)
+        {
+            this.deltaTable = deltaTableHandle.getDeltaTable();
+            this.fileIterator = DeltaExpressionUtils.iterateWithPartitionPruning(
```

Comment:
`DeltaExpressionUtils.iterateWithPartitionPruning` has the config check. 

The reason for not doing partition pruning in Metadata.getTableLayouts is: Delta partition values are are stored for each file in a separate entry in the delta log. Finding the list of the files in Metadata is an expensive operation. Instead of the pruning is done when generating the splits where the pruning is done for fixed set of files at a time.

Reply:
Removed the config property. No longer relevant.